### ISSUE TITLE
Check on Presentation display availability in getAvailability can never be true

### ISSUE THREAD

**Issue by tidoust** [w3c/presentation-api#388]
One last one :) My apologies for not seeing that before as well!

[`getAvailability`](http://w3c.github.io/presentation-api/#getting-the-presentation-displays-availability-information) is the only algorithm that sets the [presentation display availability](http://w3c.github.io/presentation-api/#dfn-presentation-display-availability). That algorithm also creates (and never resets) the [presentation availability promise](http://w3c.github.io/presentation-api/#dfn-presentation-availability-promise).

The presentation display availability will always be `null` the first time `getAvailability` is called so condition in step 7 that checks on the presentation display availability will always evaluate to `false`. The presentation availability promise will always be set in subsequent calls to `getAvailability`, meaning that the algorithm will stop at step 2, return that promise, and never reach step 7.

In short, unless I missed something, I believe that step (and substeps) can be dropped.

This did not happen before because there was no such thing as a "presentation availability promise".

**Comment by mfoltzgoogle** [comment]
Sorry, I don't follow.   The availability will be set to true if there is a display in the _list of available presentation displays_ that is compatible, found earlier by the monitoring steps.

**Comment by tidoust** [comment]
Well, the step 7 I was looking at is now step 6: "If the presentation display availability for presentationRequest is not null, then: [...]".

The problem boils down to whether the user agent can reset the _presentation availability promise_ at some point in time, as I alluded to in a comment on PR #390: https://github.com/w3c/presentation-api/pull/390#issuecomment-264830700

If it cannot (which is how I read the current algorithm), then the promise will be set in any further call to `getAvailability`. That means any further call will stop at step 2, and so step 6. is useless.

Now, if the user agent can garbage collect the _presentation availability promise_ once it is settled and/or no longer referenced, then step 6. remains useful!

That step would also be useful if `start` can set the _presentation display availability_, as envisioned in PR #390.

**Comment by mfoltzgoogle** [comment]
Okay, I am starting to understand what you are getting at.  Thanks for your patience.

The question is whether the availability promise/object is tied to the lifetime of the request that created them, which means they are created once and only once.  Or, if they can be GCed independently from the request and recreated later.

I believe the intention is that the developer can use the lifetime of the availability object to control when the browser is engaged in continuous monitoring of displays, and responsible usage would discard the object when the site isn't offering a UX to start presentation.

We didn't add a specific `discard()` or `dispose()` API for this; that doesn't seem to be a common Web idiom.

Unless we go that route, GC should be able to discard the availability object/promise independent of the request, and in that case, we would need to recreate them if getAvailability() is called again.

@schien Does that make sense?  Any other input?

**Comment by tidoust** [comment]
> Okay, I am starting to understand what you are getting at. Thanks for your patience.

No problem, I can tell my comments are getting more and more convoluted :)

> Unless we go that route, GC should be able to discard the availability object/promise independent of the request, and in that case, we would need to recreate them if getAvailability() is called again.

OK, this issue becomes moot in that case.
See proposal in https://github.com/w3c/presentation-api/pull/390#issuecomment-265115507 to clarify GC expectations.

**Comment by mfoltzgoogle** [comment]
I believe this should be addressed by PR #387.
